### ISSUE TITLE
Fontconfig not finding fonts under /mingwXX/share/fonts

### ISSUE THREAD

**Issue by BrisingrAerowing** [msys2/MINGW-packages#5762]
I've added a few fonts to the mentioned directory (in subdirectories, like with the Adobe Source Code Pro package), and even after running fc-cache -rs, the fonts don't show up in any app I've tried (Inkscape, Gimp and Geany so far, all from official MSYS2 packages). Even the source code pro fonts don't show up. I tested putting fonts directly under the fonts directory, and that didn't work either.

**Comment by BrisingrAerowing** [comment]
This is possibly fixable by using the 'use_fontconfig' meson option Pango has, as that seems to be the root of the issue. I'll test this with a custom build in a moment.

**Comment by BrisingrAerowing** [comment]
It seems this requires a newer Pango version.

**Comment by liZe** [comment]
For example, Fontconfig doesn’t find fonts installed by the mingw-w64-x86_64-ttf-dejavu package. That’s strange, because the dejavu package calls `fc-cache`, so we can imagine that it used to work.

**Comment by BrisingrAerowing** [comment]
Yes. By default Pango uses the Windows font system. It can be forced to use Fontconfig in code, and newer versions allow forcing Fontconfig by default. The version used in Msys2 does not support that, though.

**Comment by liZe** [comment]
> Yes. By default Pango uses the Windows font system. It can be forced to use Fontconfig in code, and newer versions allow forcing Fontconfig by default. The version used in Msys2 does not support that, though.

I’ve made Pango use Fontconfig with Msys2, and it works well :smile:.

The only problem I have is that the DejaVu fonts from the package are installed in a folder that’s not scanned by Fontconfig (they’re in `mingwXX/share/fonts`). Copying the fonts into `~/.fonts` for example fixes the problem. I suppose that adding the folder to the Fontconfig configuration file would work too.

**Comment by BrisingrAerowing** [comment]
How?

**Comment by liZe** [comment]
> How?

You can look at [how it’s done in WeasyPrint](https://github.com/Kozea/WeasyPrint/blob/079f21251d886918bf99b71bc02f2da39dcb3e68/weasyprint/fonts.py#L254). It basically uses a simple Python binding, the same C code should work as well.

This code works well with Msys2 as we use it for WeasyPrint’s CI (some tests rely on Fontconfig).

**Comment by podsvirov** [comment]
I confirm. I also noticed this behavior when creating #6381.

**Comment by lazka** [comment]
Can anyone test again with the latest packages?

**Comment by liZe** [comment]
> Can anyone test again with the latest packages?

It doesn’t work for me.

**Comment by liZe** [comment]
> Maybe related to #6661

It’s strange, because it really looks like the same bug. But even with `mingw-w64-x86_64-fontconfig-2.13.92-2` my fresh fontconfig installation doesn’t find fonts installed by `mingw-w64-x86_64-ttf-dejavu` if they’re not copied into `~/.fonts`.

**Comment by naveen521kk** [comment]
does https://github.com/msys2/MINGW-packages/issues/7230#issuecomment-753256864 help?

**Comment by liZe** [comment]
> does [#7230 (comment)](https://github.com/msys2/MINGW-packages/issues/7230#issuecomment-753256864) help?

It doesn’t help because I force Pango to use Fontconfig, even on Windows.

Adding the fonts to `~/.fonts` (which is a default Fontconfig path) works, meaning that Fontconfig is used and works, but for some reason doesn’t look into `/mingwXX/share/fonts`.
